src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/FactsMapper.java

```diff
@@ -210,23 +165,33 @@ private FactHearingType mapDbHearingTypeToFactHearingType(HearingType ht) {
         return new FactHearingType(ht.getCode(), ht.getDescription());
     }
 
-    public String mapHearingPartToRuleJsonMessage(HearingPart hearingPart) throws JsonProcessingException {
-        Hearing hearing = hearingPart.getHearing();
+    public String mapHearingPartToRuleJsonMessage(HearingPart hearingPart) {
         FactHearingPart factHearingPart = new FactHearingPart();
+        Hearing hearing = hearingPart.getHearing();
 
         factHearingPart.setId(hearingPart.getId().toString());
-        factHearingPart.setDuration(hearing.getDuration());
+
+        if (hearing.isMultiSession() && hearingPart.getSession() != null) {
```

Comment:
I don't see any exclusion of deleted parts or parts based on status_config.count_in_utilisation.
Is it done somewhere else?

Reply:
it's a good point. We haven't had an idea of status at the point when unlist was developed. And when there were added nobody add such filtering to do not send such hp to rules. I think we need a separate story for tinding up facts that are sent to rules.